src/openrct2/network/NetworkServerAdvertiser.cpp

```diff
@@ -108,8 +108,12 @@ class NetworkServerAdvertiser final : public INetworkServerAdvertiser
         json_t* body = json_object();
         json_object_set_new(body, "key", json_string(_key.c_str()));
         json_object_set_new(body, "port", json_integer(_port));
-        request.body = json_dumps(body, JSON_COMPACT);
+
+        char* bodyDump = json_dumps(body, JSON_COMPACT);
+        request.body = bodyDump;
         request.header["Content-Type"] = "application/json";
+        free(bodyDump);
```

Comment:
It does not seem correct to me to free `request.body` before the actual request is done. If anything, it should be freed in the lambda expression in the `Http::DoAsync` call below.

Reply:
It's not necessary indeed, Sijmen probably assumed that `request.body` was a pointer.